src/effects/effectsbackend.cpp

```diff
@@ -45,15 +45,17 @@ bool EffectsBackend::canInstantiateEffect(const QString& effectId) const {
     return m_registeredEffects.contains(effectId);
 }
 
-EffectPointer EffectsBackend::instantiateEffect(EffectsManager* pEffectsManager,
-                                                const QString& effectId) {
+bool EffectsBackend::instantiateEffect(EffectsManager* pEffectsManager,
+        const QString& effectId, const EffectSlotPointer pEffectSlot,
```

Comment:
I'm confused why this function takes an EffectSlotPointer now. Shouldn't it return an EngineEffect which then gets loaded into the EffectSlot?